### PR TITLE
Allow to construct a SparseArray from arrays and onsets

### PR DESCRIPTION
**Pull request by thequilo**

### CODE REVIEW

File: paderbox/array/sparse.py [hunk 1]
```diff
@@ -397,16 +408,51 @@ def from_array_and_onset(cls, array, onset, shape):
         array([1., 1., 0., 0., 0., 0., 0., 0., 0., 0.])
         >>> SparseArray.from_array_and_onset(np.ones(5), 7, 10)
         SparseArray(_SparseSegment(onset=7, array=array([1., 1., 1.])), shape=(10,))
+        >>> SparseArray.from_array_and_onset(np.ones(5), 7).as_contiguous()
+        array([0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1.])
         """
+        if shape is None:
+            shape = list(array.shape)
+            shape[0] += onset
```
Comment (boeddeker) [inline paderbox/array/sparse.py:416]: change to -1

File: paderbox/array/sparse.py [hunk 2]
```diff
@@ -397,16 +408,51 @@ def from_array_and_onset(cls, array, onset, shape):
...
+            if shape[0] < 0:
+                shape[0] = 0
         out = SparseArray(shape=shape)
-        if onset < 0:
-            array = array[..., -onset:]
-            onset = 0
-        if array.shape[-1] + onset > out.shape[-1]:
-            array = array[..., :out.shape[-1] - onset]
-        out._add_segment(_SparseSegment(onset, array))
+        out._add_segment(_SparseSegment(onset, array).get_clipped(out.shape[-1]))
```
Comment (boeddeker) [inline paderbox/array/sparse.py:420]: Should we disable, that _SparseSegment could represent a segment, that is partially outside?

File: paderbox/array/sparse.py (hunk 1 above)
Comment (thequilo) [reply paderbox/array/sparse.py]: Done

File: paderbox/array/sparse.py (hunk 2 above)
Comment (thequilo) [reply paderbox/array/sparse.py]: Added a method  `_SparseSegment.from_array` that constructs a clipped segment. This should be used everywhere where a `_SparseSegment` is constructed from an array that is not controlled by a SparseArray

File: paderbox/array/sparse.py [hunk 3]
```diff
@@ -397,16 +397,51 @@ def from_array_and_onset(cls, array, onset, shape):
...
+        >>> SparseArray.from_arrays_and_onsets([np.ones(2), np.ones(1)], [0, 3]).as_contiguous()
+        array([1., 1., 0., 1.])
+        >>> SparseArray.from_arrays_and_onsets([np.ones(2), np.ones(1)], [0, 3], (10,)).as_contiguous()
+        array([1., 1., 0., 1., 0., 0., 0., 0., 0., 0.])
+        """
+        if shape is None:
+            for array, onset in zip(arrays, onsets):
+                if shape is None or array.shape[-1] + onset > shape[-1]:
+                    shape = list(array.shape)
+                    shape[-1] += onset
```
Comment (boeddeker) [inline paderbox/array/sparse.py:424]: Suggestion: Add helper function, that predicts the shape.

Why? `from_array_and_onset` has
```
            if shape[-1] < 0:
                shape[-1] = 0
```
while it is missing in `from_arrays_and_onsets`.

File: paderbox/array/sparse.py [hunk 4]
```diff
@@ -397,16 +397,51 @@ def from_array_and_onset(cls, array, onset, shape):
...
+
+        out = SparseArray(shape=shape, _pad_value=_pad_value_like(arrays[0], pad_value))
+
+        assert len(arrays) == len(onsets)
```
Comment (boeddeker) [inline paderbox/array/sparse.py:428]: `assert len(arrays) == len(onsets), (len(arrays), len(onsets))`

File: paderbox/array/sparse.py [hunk 5]
```diff
@@ -397,16 +397,51 @@ def from_array_and_onset(cls, array, onset, shape):
...
+
+        for array, onset in zip(arrays, onsets):
+            out._add_segment(_SparseSegment.from_array(array, onset, out.shape[-1]))
+
         return out
 
+    def as_arrays_and_onsets(self):
```
Comment (boeddeker) [inline paderbox/array/sparse.py:435]: Add a small sentence with a motivation for this function?

**Pull request merged**